coalib/settings/FunctionMetadata.py

```diff
@@ -2,20 +2,22 @@
 from copy import copy
 from inspect import getfullargspec, ismethod
 
+from coalib.misc.Decorators import enforce_signature
 from coalib.settings.DocumentationComment import DocumentationComment
 
 
 class FunctionMetadata:
     str_nodesc = "No description given."
     str_optional = "Optional, defaults to '{}'."
 
+    @enforce_signature
```

Comment:
I have no clue why this is needed. Can you add more info in the commit message ?